crates/core_arch/src/x86/avx512f.rs

```diff
@@ -87,6 +87,32 @@ pub unsafe fn _mm512_setr_epi32(
     transmute(r)
 }
 
+/// Gather 64-bit integers from memory using 32-bit indices.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm512_i32gather_epi64)
+#[inline]
+#[target_feature(enable = "avx512f")]
+#[cfg_attr(test, assert_instr(vpgatherdq))]
+pub unsafe fn _mm512_i32gather_epi64(offsets: __m256i, slice: *const u8, scale: i32) -> __m512i {
+    let zero = _mm512_setzero_si512().as_i64x8();
```

Comment:
You should use `_mm512_undefined` here instead to match what Clang is doing.

Reply:
Hmm actually it seems that Clang defines `_mm512_undefined` as zero-initialization, so it doesn't matter either way.

Reply:
Are you sure? I see it defined as a particular builtin, but `_mm512_setzero` is explicitly defined as zero initialization. I'm not sure of the behavior of `__builtin_ia32_undef512`,  however.

https://github.com/llvm/llvm-project/blob/1b02db52b79e01f038775f59193a49850a34184d/clang/lib/Headers/avx512fintrin.h#L189
https://github.com/llvm/llvm-project/blob/a3dc9490004ce1601fb1bc67cf218b86a6fdf652/clang/include/clang/Basic/BuiltinsX86.def#L40
https://github.com/llvm/llvm-project/blob/1b02db52b79e01f038775f59193a49850a34184d/clang/lib/Headers/avx512fintrin.h#L259
https://github.com/llvm/llvm-project/blob/1b02db52b79e01f038775f59193a49850a34184d/clang/lib/Headers/avx512fintrin.h#L253

LLVM _should_ be able to optimize away the dead store, but I'm happy to change the code regardless. I'm not quite sure how/if I can implement `_mm512_undefined` since my reading of the `std::mem::MaybeUninit` is that I couldn't create an unitialized `__m512i` without inviting UB. Assuming the calling convention allows it, I should be able to create a `MaybeUninit<__m512i>` and pass that to `vpgatherdq`.